### PR TITLE
tool/lgc: use new pass manager for the middle-end pipeline

### PR DESCRIPTION
**Pull request by nhaehnle**
Note that the new pass manager is already used by default for LLPC.

### CODE REVIEW

File: lgc/patch/Patch.cpp [hunk 1]
```diff
@@ -247,7 +247,7 @@ void Patch::addPasses(PipelineState *pipelineState, lgc::PassManager &passMgr, b
 //
 // @param [in/out] passMgr : Pass manager
 void Patch::registerPasses(lgc::PassManager &passMgr) {
-#define LLPC_PASS(NAME, CLASS) passMgr.registerPass(NAME, decltype(CLASS)::name());
+#define LLPC_PASS(NAME, CLASS) passMgr.registerPass(NAME, CLASS::name());
```
Comment (bsaleil) [inline lgc/patch/Patch.cpp:250]: We should do the same thing in `llpcSpirvLower.cpp` and also update `llpc/lower/PassRegistry.inc` for consistency. We also need to update the doc (`llpc/docs/AddingPasses.md`).

Comment (nhaehnle) [reply lgc/patch/Patch.cpp:250]: You're right, I've updated the PR.

**Comment by amdvlk-admin** [comment]
### Test summary for commit ecb8b9459c1e1e4c1b5746f23cd7d8d9a7144184
<details><summary>CTS tests (Failed: 0/216210)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35449/73547 (48.2%)</li>
<li>Failed: 0/73547 (0.0%)</li>
<li>Not Supported: 38098/73547 (51.8%)</li><li>Warnings: 0/73547 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 33142/69116 (48.0%)</li>
<li>Failed: 0/69116 (0.0%)</li>
<li>Not Supported: 35974/69116 (52.0%)</li><li>Warnings: 0/69116 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37710/73547 (51.3%)</li>
<li>Failed: 0/73547 (0.0%)</li>
<li>Not Supported: 35837/73547 (48.7%)</li><li>Warnings: 0/73547 (0.0%)</li>
</ul></details>


</ul></details>

**Review by bsaleil — APPROVED** [review]
LGTM, thanks

**Review by amdrexu — APPROVED** [review]

**Pull request merged**